src/utils/api.js

```diff
@@ -0,0 +1,34 @@
+async function fetchApiCall(url, method, body = null) {
+    try {
+      const headers = {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+        'Authorization': 'Token 123456789abcdefg',
+      }
+  
+      const config = {
+        method: method,
+        headers: headers,
+        body: body ? JSON.stringify(body) : null
+      }
+
+      const response = await fetch(url, config)
+  
+      if (!response.ok) {
+        try {
+          const errorData = await response.json()
+          throw new Error(errorData.message || 'Error fetching data')
+        } catch (jsonError) {
+          throw new Error('Error fetching data')
+        }
+      }
+  
+      return await response.json()
+    } catch (error) {
+      console.error('Fetch API Call Error:', error)
+      throw error
+    }
+  }
+  
+  export { fetchApiCall }
+  
```

Comment:
Nice refactor here! just missing a newline

Reply:
Isn't line 34 a new line?

Reply:
It's hard to say. POSIX standard requires a `\n` at the end of file (which VSCode can insert for you) for many unix tools to properly display/interpret these files. That blank line may be something different than the standard, and GitHub is showing that the file isn't properly terminated.

Reply:
hmm on my file it does end with an extra line. Is that what it is? I can send you screenshots